RakutenShop/Scenes/RecipeDetailView/RecipeDetailViewController.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  RecipeDetailViewController.swift
+//  RakutenShop
+//
+//  Created by Garnik Ghazaryan on 2/11/20.
+//  Copyright © 2020 Garnik Ghazaryan. All rights reserved.
+//
+
+import UIKit
+import ReactiveCocoa
+import ReactiveSwift
+import AlamofireImage
+import SVProgressHUD
+
+class RecipeDetailViewController: BaseViewController {
+
+    @IBOutlet weak var recipeImageView: UIImageView!
+    @IBOutlet weak var instructionsTableView: UITableView!
+    @IBOutlet weak var similarRecipesCollectionView: UICollectionView!
+    @IBOutlet weak var caloriesLabel: UILabel!
+    @IBOutlet weak var carbsLabel: UILabel!
+    @IBOutlet weak var fatsLabel: UILabel!
+    @IBOutlet weak var proteinsLabel: UILabel!
+    
+    var viewModel: RecipeDetailViewViewModel?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        instructionsTableView.dataSource = self
+        instructionsTableView.delegate = self
+        instructionsTableView.tableFooterView = UIView()
+        
+        similarRecipesCollectionView.dataSource = self
+        similarRecipesCollectionView.delegate = self
+        
+        // Do any additional setup after loading the view.
+    }
+    
+    override func setupView() {
+        let rightBarItem = UIBarButtonItem(title: "Save", style: .plain, target: self, action: #selector(saveButtonTapped))
+        self.navigationItem.setRightBarButton(rightBarItem, animated: true)
+    }
+    
+    override func setupViewModel() {
+        guard let viewModel = viewModel else { return }
+        
+        viewModel.isLoading.producer.startWithValues { (showLoading) in
+            DispatchQueue.main.async {
+                showLoading ? SVProgressHUD.show() : SVProgressHUD.dismiss()
+            }
+        }        
+        
+        viewModel.errorMessage.producer.skip(first: 1).startWithValues { (errorMessage) in
```

Comment:
I suppose we you can just start instead of startWithValues which will not take default or first value, I am not sure, but you can try it.